data/publiccloud/terraform/nvidia_gce.tf

```
@@ -0,0 +1,173 @@
+terraform {
+  required_providers {
+    google = {
+      version = "= 3.65.0"
+      source = "hashicorp/google"
+    }
+    random = {
+      version = "= 3.1.0"
+      source = "hashicorp/random"
+    }
+    external = {
+      version = "= 2.1.0"
+      source = "hashicorp/external"
+    }
+  }
+}
+
+variable "cred_file" {
+    default = "/root/google_credentials.json"
+}
+
+provider "google" {
+    credentials = var.cred_file
+    project     = var.project
+}
+
+data "external" "gce_cred" {
+    program = [ "cat", var.cred_file ]
+    query =  { }
+}
+
+variable "instance_count" {
+    default = "1"
+}
+
+variable "name" {
+    default = "openqa-vm"
+}
+
+variable "type" {
+    default = "n1-standard-2"
+}
+
+variable "region" {
+    default = "europe-west1-b"
+}
+
+variable "image_id" {
+    default = ""
+}
+
+variable "project" {
+    default = "suse-sle-qa"
+}
+
+variable "extra-disk-size" {
+    default = "1000"
+}
+
+variable "extra-disk-type" {
+    default = "pd-ssd"
+}
+
+variable "create-extra-disk" {
+    default=false
+}
+
+variable "uefi" {
+    default=false
+}
+
+variable "tags" {
+    type = map(string)
+    default = {}
+}
+
+variable "enable_confidential_vm" {
+	default=false
+}
+
+resource "random_id" "service" {
+    count = var.instance_count
+    keepers = {
+        name = var.name
+    }
+    byte_length = 8
+}
+
+resource "google_compute_instance" "openqa" {
+    count                        = var.instance_count
+    name                         = "${var.name}-${element(random_id.service.*.hex, count.index)}"
+    machine_type                 = var.type
+    zone                         = var.region
+
+    guest_accelerator {
+      type = "nvidia-tesla-t4"
+      count = 1
```

I did a diff of this file in comparison to "regular" gce.tf . this guest_accelerator is the only block which differs . I don't think it make sense to keep it as separate file . 

Just add this module into `gce.tf` with this change : 

```suggestion
      count = var.guest_accelerator ? 1 : 0
```

also add variable : 
```
variable "guest_accelerator" {
    default=false
}
```

---

i need to test this. i will come back to you

---

*ready for review*